test/test-harness.sh

```diff
@@ -34,20 +34,13 @@ docker build -t eks-optimized-ami -f "${SCRIPTPATH}/Dockerfile" "${SCRIPTPATH}/.
 overall_status=0
 
 function run(){
-    local temp_dir=$1
-    shift
-    # This variable is used to override the default value in the kubelet mock
-    KUBELET_VERSION="${KUBELET_VERSION:-}"
-    cp -f ${SCRIPTPATH}/../files/kubelet-config.json ${temp_dir}/kubelet-config.json
     docker run -v ${SCRIPTPATH}/../files/:/etc/eks/ \
-        -v ${temp_dir}/kubelet-config.json:/etc/kubernetes/kubelet/kubelet-config.json \
+        -v "$(realpath $1):/test.sh" \
         --attach STDOUT \
         --attach STDERR \
         --rm \
-        -e KUBELET_VERSION="$KUBELET_VERSION" \
-        eks-optimized-ami $@
+        eks-optimized-ami
 }
-export -f run
 
```

Comment:
This was getting a bit unwieldy; I think it's more straightforward to run the test script inside the container (and it makes assertions in the test scripts a lot simpler, without requiring volume mounts).

It also gives us an easier way to accomplish mocking on a per-test basis, instead of having a set of mocks that are used for every tests. For example, a test case that needs to mock kubelet could do:
```
> cat my-test-case.sh

function kubelet() {
  echo "my expected output"
  return $MY_EXPECTED_EXIT_CODE
}
export -f kubelet
/command/that/uses/kubelet --arg
```